src/main/java/com/netflix/imflibrary/st0377/HeaderPartition.java

```diff
@@ -1320,6 +1320,9 @@ else if(interchangeObjectBO.getClass().getEnclosingClass().equals(RGBAPictureEss
             else if(interchangeObjectBO.getClass().getEnclosingClass().equals(TimedTextDescriptor.class)){
                 essenceTypes.add(EssenceTypeEnum.SubtitlesEssence);
             }
+            else if(interchangeObjectBO.getClass().getEnclosingClass().equals(TimedTextDescriptor.class)){
```

Comment:
The loop would never reach this, as the previous if-clause is already checking for the same case. Do other sections in the code rely on the specific essence type returned here? I'd have to investigate.